vcd/resource_vcd_nsxt_security_group_test.go

```diff
@@ -0,0 +1,477 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdNsxtEdgeGatewayObject(testConfig, testConfig.Nsxt.EdgeGateway, "test-security-group-changed"),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccNsxtSecurityGroupPrereqs = testAccNsxtSecurityGroupPrereqsEmpty + `
+resource "vcd_network_routed_v2" "nsxt-backed" {
+	# This value could be larger to test more members, but left 2 for the sake of testing speed
+	count = 2
+
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+	name        = "nsxt-routed-${count.index}"
+	description = "My routed Org VDC network backed by NSX-T"
+  
+	edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+  
+	gateway       = "212.1.${count.index}.1"
+	prefix_length = 24
+  
+	static_ip_pool {
+	  start_address = "212.1.${count.index}.10"
+	  end_address   = "212.1.${count.index}.20"
+	}
+}
+
+# Create stanadlone VM to check membership
+resource "vcd_vm" "emptyVM" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+
+    power_on      = false
+	name          = "standalone-VM"
+	computer_name = "emptyVM"
+	memory        = 2048
+	cpus          = 2
+	cpu_cores     = 1
+  
+	os_type = "sles10_64Guest"
+	hardware_version = "vmx-14"
+	
+	network {
+		type               = "org"
+		name               = (vcd_network_routed_v2.nsxt-backed[0].id == "always-not-equal" ? null : vcd_network_routed_v2.nsxt-backed[0].name)
+		ip_allocation_mode = "POOL"
+		is_primary         = true
+	}
+
+	depends_on = [vcd_network_routed_v2.nsxt-backed]
+  }
+
+# Create a vApp and VM
+resource "vcd_vapp" "web" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+
+  name     = "web"
+  power_on = false
+}
+
+resource "vcd_vapp_org_network" "vappOrgNet" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+  
+	vapp_name         = vcd_vapp.web.name
+	org_network_name  = (vcd_network_routed_v2.nsxt-backed[1].id == "always-not-equal" ? null : vcd_network_routed_v2.nsxt-backed[1].name)
+
+	depends_on = [vcd_vapp.web,vcd_network_routed_v2.nsxt-backed]
+  }
+
+resource "vcd_vapp_vm" "emptyVM" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+
+  power_on      = false
+  vapp_name     = vcd_vapp.web.name
+  name          = "vapp-vm"
+  computer_name = "emptyVM"
+  memory        = 2048
+  cpus          = 2
+  cpu_cores     = 1
+
+  os_type = "sles10_64Guest"
+  hardware_version = "vmx-14"
+
+  network {
+	type               = "org"
+	name               = (vcd_vapp_org_network.vappOrgNet.id == "always-not-equal" ? null : vcd_vapp_org_network.vappOrgNet.org_network_name)
+	ip_allocation_mode = "POOL"
+	is_primary         = true
+  }
+
+  depends_on = [vcd_vapp_org_network.vappOrgNet,vcd_network_routed_v2.nsxt-backed]
+}
+`
+
+const testAccNsxtSecurityGroup = testAccNsxtSecurityGroupPrereqs + `
+resource "vcd_nsxt_security_group" "group1" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name = "test-security-group"
```

Comment:
This script is nor well formatted

Reply:
Reformatted all scripts with `terraform fmt` in this file.